src/Features/CSharp/Portable/UseCompoundAssignment/Maps.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseCompoundAssignment
+{
+    internal static class Maps
+    {
+        public static readonly ImmutableDictionary<SyntaxKind, SyntaxKind> BinaryToAssignmentMap =
+            new Dictionary<SyntaxKind, SyntaxKind>
+            {
+                { SyntaxKind.AddExpression, SyntaxKind.AddAssignmentExpression },
+                { SyntaxKind.SubtractExpression, SyntaxKind.SubtractAssignmentExpression },
+                { SyntaxKind.MultiplyExpression, SyntaxKind.MultiplyAssignmentExpression },
+                { SyntaxKind.DivideExpression, SyntaxKind.DivideAssignmentExpression },
+                { SyntaxKind.ModuloExpression, SyntaxKind.ModuloAssignmentExpression },
+                { SyntaxKind.BitwiseAndExpression, SyntaxKind.AndAssignmentExpression },
+                { SyntaxKind.ExclusiveOrExpression, SyntaxKind.ExclusiveOrAssignmentExpression },
+                { SyntaxKind.BitwiseOrExpression, SyntaxKind.OrAssignmentExpression },
+                { SyntaxKind.LeftShiftExpression, SyntaxKind.LeftShiftAssignmentExpression },
+                { SyntaxKind.RightShiftExpression, SyntaxKind.RightShiftAssignmentExpression },
+            }.ToImmutableDictionary();
+
+        public static readonly ImmutableDictionary<SyntaxKind, SyntaxKind> AssignmentToTokenMap =
```

Comment:
>AssignmentToTokenMap [](start = 75, length = 20)

Why not hard-code this second map, as you did in VB?

Reply:
VB lacks the helper function :(

Reply:
Is there an easy way to reverse a dictionary?

Reply:
The dictionary isn't reversed.  It's an entirely different mapping.  The ony difference here is that C# has a built-in function to map from AssignmentExpr kind to TokenKind.

Reply:
Sorry, I can't read apparently